tests/Unit/Evolution/Systems/Ccz4/Test_TimeDerivative.cpp

```diff
@@ -0,0 +1,1244 @@
+  Approx approx_12f = Approx::custom().epsilon(1e-11).scale(1.0);
+  for (auto& component : dt_trace_extrinsic_curvature_actual) {
+    CHECK_ITERABLE_CUSTOM_APPROX(component, zero, approx_12f);
+  }
+  // eq 12g
+  Approx approx_12g = Approx::custom().epsilon(1e-11).scale(1.0);
+  for (auto& component : dt_theta_actual) {
+    CHECK_ITERABLE_CUSTOM_APPROX(component, zero, approx_12g);
+  }
+  // eq 12h
+  // \partial_t \hat{\gamma}^i will not be 0 for KerrSchild if
+  // evolve_shift == false
+  const tnsr::I<DataVector, SpatialDim, FrameType> dt_gamma_hat_expected =
+      get_dt_gamma_hat_kerr_expected(
+          evolve_shift, inverse_conformal_spatial_metric, d_field_b);
+  Approx approx_12h = Approx::custom().epsilon(1e-11).scale(1.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(dt_gamma_hat_actual, dt_gamma_hat_expected,
+                               approx_12h);
+  // eq 12i
+  // \partial_t b will not be 0 for KerrSchild if evolve_shift == true
+  const tnsr::I<DataVector, SpatialDim, FrameType> dt_b_expected =
+      get_dt_b_kerr_expected(evolve_shift, eta, shift, d_gamma_hat, b, d_b);
+  Approx approx_12i = Approx::custom().epsilon(1e-11).scale(1.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(dt_b_actual, dt_b_expected, approx_12i);
+  // eq 12j
+  Approx approx_12j = Approx::custom().epsilon(1e-11).scale(1.0);
+  for (auto& component : dt_field_a_actual) {
+    CHECK_ITERABLE_CUSTOM_APPROX(component, zero, approx_12j);
+  }
+  // eq 12k
+  Approx approx_12k = Approx::custom().epsilon(1e-11).scale(1.0);
+  for (auto& component : dt_field_b_actual) {
+    CHECK_ITERABLE_CUSTOM_APPROX(component, zero, approx_12k);
+  }
+  // eq 12l
+  // \partial_t D_{kij} will not be 0 for KerrSchild if evolve_shift == false
+  const tnsr::ijj<DataVector, SpatialDim, FrameType> dt_field_d_expected =
+      get_dt_field_d_kerr_expected(evolve_shift, conformal_spatial_metric,
+                                   d_field_b);
+  Approx approx_12l = Approx::custom().epsilon(1e-11).scale(1.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(dt_field_d_actual, dt_field_d_expected,
+                               approx_12l);
+  // eq 12m
+  // \partial_t P_i will not be 0 for KerrSchild if evolve_shift == false
+  const tnsr::i<DataVector, SpatialDim, FrameType> dt_field_p_expected =
+      get_dt_field_p_kerr_expected(evolve_shift, d_field_b);
+  Approx approx_12m = Approx::custom().epsilon(1e-12).scale(1.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(dt_field_p_actual, dt_field_p_expected,
+                               approx_12m);
+}
+}  // namespace
+
+// Test first order CCZ4 against Minkowski and KerrSchild
+void test(const bool evolve_shift, const bool use_harmonic_slicing_condition) {
+  test_minkowski(evolve_shift, use_harmonic_slicing_condition);
+  test_kerrschild(evolve_shift, use_harmonic_slicing_condition);
+}
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.Ccz4.TimeDerivative",
+                  "[Unit][Evolution]") {
+  // Test first order CCZ4 with different settings
+  test(true, true);
```

Comment:
If it's not too difficult, changing to enums here would also be nice. E.g. `EvolveShift::Yes/No` :) You can also do for loops, not sure if that's worth it here. I'll let you decide :)
```cpp
for (const auto blah : {a, b}) {
  for (const auto bla : {c, d}) {
}}
```

Reply:
Made the enum class but I just left the testing combos without loops for now